src/main/msp/msp.c

```diff
@@ -1458,7 +1458,7 @@ static bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
                 continue;
             };
             sbufWriteU8(dst, serialConfig()->portConfigs[i].identifier);
-            sbufWriteU16(dst, serialConfig()->portConfigs[i].functionMask);
+            sbufWriteU32(dst, serialConfig()->portConfigs[i].functionMask);
```

Comment:
To prevent breaking changes we normally leave the legacy value in place and add the 32bit version to the end. Then clients that are only 16bit compatible will still work while clients that have been updated to support the 32bit mask will understand that it has been added to the message.

Reply:
That’s unfortunately not possible on this specific case. All port configs are sent at once and clients (including the configurator) expect a hardcoded port config size to determine the number of ports. 

However, since it’s gonna be a breaking change we can make it more robust by prepending the number of port and/or each member size at the start of the payload. That will let clients read up to the point where they understand and ignore the rest. What do you think about that?

Reply:
We may have to deprecate the `MSP_CF_SERIAL_CONFIG` message (and related) and leave it the way it is for legacy clients and add new messages that are more flexible. A "breaking" change would be very painful. But it sounds like it's going to need quite a bit more discussion.

Reply:
There's a comment in `msp_protocol.h` stating:

```c
// private - only to be used by the configurator, the commands are likely to change
#define MSP_CF_SERIAL_CONFIG            54
#define MSP_SET_CF_SERIAL_CONFIG        55
```
While a breaking change is always problematic, one could say that clients relying on these two messages have been warned.

On the other hand, there's a `MSP2_COMMON_SERIAL_CONFIG` in INAV, but its payload is almost the same as `MSP_CF_SERIAL_CONFIG` except for the `functionMask` field, which uses 32 bits instead of 16. I don't like this solution because if we ever need to make more changes to the data, we're left on the same spot. Any payload format that sends multiple elements without specifying a count or an element size is impossible to extend gracefully (and unfortunately, both BF and INAV still have quite a few of those).

However, since I doubt there are many clients implementing `MSP2_COMMON_SERIAL_CONFIG` besides the INAV configurator, it might be a good idea to introduce the breaking change there by prepending the port count. This could be represented by a single byte at the start and is very unlikely that we'd ever support targets with more than 255 serial ports. Then we can implement the same format for BF, which is always beneficial.

Reply:
@etracer65 I've proposed changing `MSP2_COMMON_SERIAL_CONFIG` in INAV and no one objected, so let's see if we can agree on a common format and leave `MSP_CF_SERIAL_CONFIG` as is.
 
The payload in pseudo-c-ish would look like

```c
struct port_info {
    uint8_t identifier;
    uint32_t functionMask;
    uint8_t mspBaudrateIndex;
    uint8_t gpsBaudrateIndex;
    uint8_t telemetryBaudrateIndex;
    uint8_t peripheralBaudrateIndex;
} __attribute__((packed));

struct payload {
    uint8_t port_count;
    struct port_info ports[port_count];
} __attribute__((packed));
```

This would let us add more fields per port in the future, since callers would be able to determine the port info size in the payload from `port_count` and read only the bytes they understand from each port. What do you think about that?

Reply:
Yeah, I think yielding to a new command (`MSP2_COMMON_SERIAL_CONFIG` seems to be a good candidate) while leaving the legacy command unchanged, and structuring all new commands such that all variable length elements are prepended with a count to make them extensible is the right move here.